mpcite/utility.py

```diff
@@ -0,0 +1,349 @@
+from mpcite.models import ConnectionModel, ELinkPostResponseModel, ELinkGetResponseModel, DOIRecordModel, \
+    ExplorerGetJSONResponseModel, ElinkResponseStatusEnum
+from abc import abstractmethod, ABCMeta
+from typing import Union, List, Dict
+import logging
+import requests
+from urllib3.exceptions import HTTPError
+from xmltodict import parse
+from dicttoxml import dicttoxml
+from xml.dom.minidom import parseString
+import json
+from tqdm import tqdm
+
+
+class Adapter(metaclass=ABCMeta):
+    def __init__(self, config: ConnectionModel):
+        self.config = config
```

Comment:
Why not integrate the Config as part of the Adapter?

Reply:
i was thinking that there might be more sources that we wanted to push out, so currently we have three -- elink, explorer, and elsevier, but if we want one, we just need to make an adapter, that implements the GET and POST methods. 

this initialization would make it easier to initialize that individual adapter